main.ts

```diff
@@ -8,153 +8,161 @@ import fs = require("fs");
+    if (username) {
+      form.append("username", username);
+      console.log("DAST username set to: " + username);
     }
-    if (files.length > maxUploadFiles) {
-      throw new Error(
-        "Too many files (" + files.length + ") match the provided glob pattern; please write a more restrictive pattern to match no more than " + maxUploadFiles + " files."
-      );
+    if (password) {
+      form.append("password", password);
+      console.log("DAST password is set to: (hidden)");
+    }
+    if (comments) {
+      form.append("comments", comments);
+      console.log("Comments are set to: " + comments);
+    }
+    if (release_id) {
+      form.append("release_Id", release_id);
+      console.log("Release ID is set to: " + release_id);
+    }
+    if (platform_variant) {
+      form.append("platform_variant", platform_variant);
+      console.log("Platform variant is set to: " + platform_variant);
+    }
+    if (external_id) {
+      form.append("external_id", external_id);
+      console.log("External ID is set to: " + external_id);
     }
-    
-    console.log("Found " + files.length + " files to upload.");
-
-    // Upload all the files that matched the file path
-    let file_idx: number = 1;
-    let output: Array<any> = []
-    for (const file_path of files) {
-      if (!fs.existsSync(file_path)) {
-        throw new Error("Could not access file:" + file_path);
-      }
 
-      console.log("Processing file " + file_path + " (" + file_idx + " of " + files.length + ").");
+    // retry upload maxRetries times
+    for (let loop_idx = 0; loop_idx < maxRetries; loop_idx++) {
 
-      const form = new FormData();
-      form.append("file", fs.createReadStream(file_path));
-      // only append optional fields if explicitly set 
-      if(username) {
-        form.append("username", username);
-        console.log("DAST username set to: " + username);  
-      }
-      if(password) {
-          form.append("password", password);
-          console.log("DAST password is set to: (hidden)");
-      }
-      if(comments) {
-          form.append("comments", comments);
-          console.log("Comments are set to: " + comments);
-      }
-      if(release_id) {
-          form.append("release_Id", release_id);
-          console.log("Release ID is set to: " + release_id);
-      }
-      if(platform_variant) {
-          form.append("platform_variant", platform_variant);
-          console.log("Platform variant is set to: " + platform_variant);
+      // Send the auth request to get the upload URL
+      const auth_response = await upload_step_init(dt_upload_api_key);
+
+      let auth_json;
+      try {
+        auth_json = await auth_response.json();
+      } catch (err) {
+        core.setFailed(err);
       }
-      if(external_id) {
-          form.append("external_id", external_id);
-          console.log("External ID is set to: " + external_id);
+
+      if (auth_response.status !== 200) {
```

Comment:
The retry system is very flawed here, for instance here if the function returns something other than a `200` it will not be retried.